tools/sourcemap-tools/src/helpers/stringHelpers.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Appends `value` to `str` before trailing whitespaces in `str`.
+ * @param str String to append to.
+ * @param value String to append.
+ * @example
+ * const str = 'abc\n\n';
+ * const value = 'def';
+ * const appended = appendBeforeWhitespaces(str, value);
+ */
+export function appendBeforeWhitespaces(str: string, value: string) {
+    const whitespaces = str.match(/\s*$/)?.[0];
+    if (!whitespaces) {
+        return str + value;
+    }
+
+    return str.substring(0, whitespaces.length) + value + whitespaces;
```

Comment:
I may not be understanding this correctly, but do we want to use `whitespaces.length` here? Wouldn't we want str without the whitespaces + value + whitespaces?

Reply:
Yeah, I screwed that up. Thanks for the heads-up.

I've added some unit tests.